src/gen/conditions.rs

```diff
@@ -123,6 +126,18 @@ fn parse_args(
                 range_cache,
                 flags,
             )?;
+            // CREATE_COIN takes an optional 3rd parameter, which is a 32 byte
+            // hash. If we find anything else, that's still OK, since garbage is
+            // ignored.
+            if let Ok(c) = rest(a, c) {
+                if let Ok(param) = first(a, c) {
+                    if let SExp::Atom(b) = a.sexp(param) {
+                        if a.buf(&b).len() == 32 {
```

Comment:
How about this?
```suggestion
                        if a.buf(&b).len() <= 32 {
```

Shouldn't be much worse, and then supports shorter hints too.

Reply:
I guess we don't have to be strict on length being exactly 32 bytes, less or equal gives some extra flexibility to clients so let's go with it

Reply:
One consequence of making this proposed change is that an *empty* atom (as long as it isn't the canonical `null` value) will still count as a *set* hint. but the canonical `null` won't. That's quite unintuitive. So, if we want hints to be any size, an empty atom should probably not count as a hint.

Reply:
that makes sense